lib/mlir/Dialect/Clift/IR/CliftAttributes.cpp

```diff
@@ -291,12 +294,20 @@ static AttrType parseImpl(mlir::AsmParser &parser, llvm::StringRef TypeName) {
   using FieldsParserType = ::mlir::FieldParser<FieldsVectorType>;
   ::mlir::FailureOr<FieldsVectorType> Fields(FieldsVectorType{});
 
-  if (parser.parseOptionalRSquare().failed()) {
+  const auto ParseFieldsRSquare = [&]() -> bool {
```

Comment:
Can you be more verbose in the commit message about what's being fixed (commit "Fix Clift type parsing ")?

Reply:
I added this explanation:
* Allow empty parameter list for FunctionAttr.
* Allow empty field list for struct but not union.